sentry_sdk/integrations/spark/spark_driver.py

```diff
@@ -0,0 +1,174 @@
+from sentry_sdk import configure_scope
+from sentry_sdk.hub import Hub
+from sentry_sdk.integrations import Integration
+
+
+class SparkIntegration(Integration):
+    identifier = "spark"
+
+    @staticmethod
+    def setup_once():
+        # type: () -> None
+        patch_spark_context_init()
+
+
+def patch_spark_context_init():
+    # type: () -> None
+    from pyspark import SparkContext  # type: ignore
+    from pyspark.java_gateway import ensure_callback_server_started  # type: ignore
+
+    spark_context_init = SparkContext._do_init
+
+    def _sentry_patched_spark_context_init(self, *args, **kwargs):
+        init = spark_context_init(self, *args, **kwargs)
+
+        if not getattr(self, "_sentry_is_patched", False):
+            # So workers have access to app_name and id
+            self.setLocalProperty("app_name", self.appName)
+            self.setLocalProperty("application_id", self.applicationId)
+
+            # Initialize Sentry Listener to start recording breadcrumbs
+            hub = Hub.current
+            gw = self._gateway
+            ensure_callback_server_started(gw)
+            listener = SentryListener(hub)
```

Comment:
Instead of passing the hub around, would it be possible to access `Hub.current` in each listener method?